src/policy_provider/SamplePolicyProvider.c

```diff
@@ -408,11 +420,16 @@ int BSLP_PolicyRule_Init(BSLP_PolicyRule_t *self, const char *desc, BSLP_PolicyP
 {
     ASSERT_ARG_NONNULL(self);
     memset(self, 0, sizeof(*self));
-    string_init_set_str(self->description, desc);
-    self->sec_block_type    = sec_block_type;
-    self->target_block_type = target_block_type;
-    self->predicate         = predicate;
-    self->context_id        = context_id;
+
+    size_t desc_sz    = strnlen(desc, BSLP_POLICYPREDICATE_ARRAY_CAPACITY);
+    self->description = BSL_MALLOC(desc_sz + 1);
+    strncpy(self->description, desc, desc_sz);
+    self->description[desc_sz - 1] = '\0';
```

Comment:
This index is off by one, the sz value just needs to be used consistently. Either it is the strlen (excluding null terminator) or it is size including terminator.